### ISSUE TITLE
Classify training, pretrained weights are not correctly transferred as expected

### ISSUE THREAD

**Issue by kingsley718** [ultralytics/ultralytics#2196]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

The argument "pretrained" is forced to be "False".  So any passed pretrained model weights will be reset to default parameters.

https://github.com/ultralytics/ultralytics/blob/d0b616e41e827e1ac65ee99da1339e707a984989/ultralytics/yolo/v8/classify/train.py#L31-L34

The following code may solve the problem.
` pretrained = self.args.pretrained`

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Laughing-q** [comment]
@AyushExel From the code it looks the user is right, any thoughts?

**Comment by AyushExel** [comment]
@Laughing-q yeah pretrained should not be hardcoded. I don't remember if we did this for any specific reason or it's just a typo. 
@glenn-jocher does this ring a bell ?

**Comment by Laughing-q** [comment]
@AyushExel it is from #157, probably we just incorrectly set it at the first place?🤔

**Comment by AyushExel** [comment]
Okay. I'll fix it then shortly

**Comment by Laughing-q** [comment]
@AyushExel @kingsley718 hey guys, actually I just noticed this `reset_parameters` will not be called at all when you training from `.pt` file since `get_model` is called only when the model is `.yaml` file. So no need to worry about it. :)
[https://github.com/ultralytics/ultralytics/blob/efc941aa8131eac680872edf202b8c23be176abd/ultralytics/yolo/v8/classify/train.py#L63-L64](https://github.com/ultralytics/ultralytics/blob/efc941aa8131eac680872edf202b8c23be176abd/ultralytics/yolo/v8/classify/train.py#L63-L64)
[https://github.com/ultralytics/ultralytics/blob/efc941aa8131eac680872edf202b8c23be176abd/ultralytics/yolo/v8/classify/train.py#L27-L36](https://github.com/ultralytics/ultralytics/blob/efc941aa8131eac680872edf202b8c23be176abd/ultralytics/yolo/v8/classify/train.py#L27-L36)

**Comment by kingsley718** [comment]
@Laughing-q 
Based on my testing, `get_model` is always called by the trainer. Meanwhile, `setup_model` is only called by the predictor.

https://github.com/ultralytics/ultralytics/blob/efc941aa8131eac680872edf202b8c23be176abd/ultralytics/yolo/engine/model.py#L367

https://github.com/ultralytics/ultralytics/blob/efc941aa8131eac680872edf202b8c23be176abd/ultralytics/yolo/engine/model.py#L249

**Comment by Laughing-q** [comment]
@kingsley718 Thanks for reminding! Yes you're right! I tested it with:
```python
trainer = classify.ClassificationTrainer()
trainer.train()
```
and this case `get_model` would not be called if you use a pt file.
But yes for the use case `model.train()`, it'll be called.

@AyushExel Then the issue still exists.😂

**Comment by glenn-jocher** [comment]
@kingsley718 thanks for letting us know. I've added a TODO to work on this.

**Comment by glenn-jocher** [comment]
@kingsley718 @Laughing-q Great news 😃! Your original issue may now be resolved ✅ in PR #3239. This PR connects the pretrained usage here to the argument value and defaults the argument value to True. Now all classify trainings will start from pretrained models by default, and can be reverted to non-pretrained with `pretrained=False`. Removing TODO as completed.

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
